tests/core/test_asan_memchr.c

```diff
@@ -1,5 +1,5 @@
 #include <string.h>
 
 int main() {
-  return (int)memchr("hello", 'z', 7);
+  return (int)(long)memchr("hello", 'z', 7);
```

Comment:
Why do two casts here? (happens in a few more tests too)

Reply:
because casting a 64-bit pointer straight to `int` results in a warning/error

Reply:
I see, thanks, sgtm.

Reply:
I would use `intptr_t` here .. it should work with just a single cast I think.

Reply:
@sbc100 how? `intptr_t` is 64-bit.. that won't let itself be passed to an `int` context without warning/error.